CMakeLists.txt

```diff
@@ -227,7 +227,15 @@ target_compile_features(test_vn PUBLIC cxx_std_11)
 ### system executable ###
 ###                   ###
 
-add_executable(alpha_build drivebrain_app/main.cpp)
+add_executable(alpha_build 
+    drivebrain_app/main.cpp
+    drivebrain_app/DriveBrainApp.cpp
+)
+
+# Add include directory for DriveBrainApp header
+target_include_directories(alpha_build PRIVATE
+    ${CMAKE_CURRENT_SOURCE_DIR}/drivebrain_app
```

Comment:
target_include_directories should specify both the local build and installed include directory locations for targets like this:
```CMake
target_include_directories(drivebrain_app PUBLIC
    $<INSTALL_INTERFACE:drivebrain_app/include>
    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/drivebrain_app/include>
)

```

Reply:
also, within the drivebrain_app directory, please make an `include` and `src` folder for separating the headers and the source files.

what we should probably do is make a `drivebrain_app` library that implements everything so that we can then also link it and implement it for the testing application and the real application within the `main.cpp`

folder / file structure

- **`include/`**: header files
  - `DriveBrainApp.hpp`
- **`src/`**: run-time `.cpp`s
  - `DriveBrainApp.cpp`
  - `main.cpp`

```CMake

add_library(drivebrain_app SHARED 
    drivebrain_app/DriveBrainApp.cpp
)

target_include_directories(drivebrain_app PRIVATE
    ${CMAKE_CURRENT_SOURCE_DIR}/drivebrain_app
)

add_executable(alpha_build
    drivebrain_app/main.cpp
)

target_link_libraries(alpha_build PRIVATE
    drivebrain_app
)


```